pkg/utils/validator/clusterresourceplacement.go

```diff
@@ -39,8 +39,8 @@ var (
 	uniqueTolerationErrFmt       = "toleration %+v already exists, tolerations must be unique"
 
 	// Below is the map of supported capacity types.
-	supportedResourceCapacityTypesMap = map[string]bool{propertyprovider.TotalCapacityName: true, propertyprovider.AllocatableCapacityName: true, propertyprovider.AvailableCapacityName: true}
-	supportedResourceCapacityTypes    = reflect.ValueOf(supportedResourceCapacityTypesMap).MapKeys()
+	supportedResourceCapacityTypesMap = map[string]bool{propertyprovider.AllocatableCapacityName: true, propertyprovider.AvailableCapacityName: true, propertyprovider.TotalCapacityName: true}
+	resourceCapacityTypes             = supportedResourceCapacityTypes()
```

Comment:
why creating this alias?

Reply:
This var gets created once for this file. Since the validator function could be called many time I wanted to avoid creating the slice from the map every time the function is called.